windows-agent/cmd/ubuntu-pro-agent/agent/config.go

```diff
@@ -0,0 +1,97 @@
+package agent
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/canonical/ubuntu-pro-for-wsl/common"
+	log "github.com/canonical/ubuntu-pro-for-wsl/common/grpc/logstreamer"
+	"github.com/canonical/ubuntu-pro-for-wsl/common/i18n"
+	"github.com/canonical/ubuntu-pro-for-wsl/windows-agent/internal/consts"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/ubuntu/decorate"
+)
+
+func initViperConfig(name string, cmd *cobra.Command, vip *viper.Viper) (err error) {
+	defer decorate.OnError(&err, "can't load configuration")
+
+	// Use command-line flag for verbosity until configuration is parsed
+	v, err := cmd.Flags().GetCount("verbosity")
+	if err != nil {
+		return fmt.Errorf("internal error: no persistent verbosity flags installed on cmd: %w", err)
+	}
+	setVerboseMode(v)
+
+	// Find a valid configuration file
+	if v, err := cmd.Flags().GetString("config"); err == nil && v != "" {
+		vip.SetConfigFile(v)
+	} else {
+		vip.SetConfigName(name)
+		vip.AddConfigPath("./")
+		vip.AddConfigPath("$HOME")
+		vip.AddConfigPath(fmt.Sprintf("$HOME/%s", common.UserProfileDir))
+		if binPath, err := os.Executable(); err != nil {
+			log.Warningf(context.Background(), "Failed to get the current executable path, not adding it as a config dir: %v", err)
+		} else {
+			vip.AddConfigPath(filepath.Dir(binPath))
+		}
+	}
+
+	// Load the config
+	if err := vip.ReadInConfig(); err != nil {
+		var e viper.ConfigFileNotFoundError
+		if errors.As(err, &e) {
+			log.Infof(context.Background(), "No configuration file: %v", e)
+		} else {
+			return fmt.Errorf("invalid configuration file: %v", err)
+		}
+	} else {
+		log.Infof(context.Background(), "Using configuration file: %v", vip.ConfigFileUsed())
+	}
```

Comment:
You could add one small test function in agent_test.go, similar to `TestNoUsageError` for example, that writes a config file setting the verbosity level to the maximum in one of the supported locations before starting the agent and looking for that file path in the captured stdout.